specification/sections/05-events.md

```diff
@@ -80,10 +80,12 @@ see: [provider events](#51-provider-events), [`provider event types`](../types.m
 
 #### Requirement 5.2.3
 
-> The `event details` **MUST** contain the `client name` associated with the event.
+> The `event details` **MUST** contain the `provider name` associated with the event.
 
-The `client name` indicates the client/provider pair with which the event is associated.
-This is especially relevant for event handlers which are attached to the `API`, not a particular client.
+The `provider name` indicates the provider from which the event originated.
+This is especially relevant for global event handlers used for general monitoring, such as alerting on provider errors. 
+
+See [setting a provider](./01-flag-evaluation.md#setting-a-provider), [creating clients](./01-flag-evaluation.md#creating-clients). 
```

Comment:
The previous requirement was problematic because it's possible for the same provider instance to be bound to multiple names... which means that an event would sometimes need to feature mutliple `client names`. I think this would be overly confusing.

The main use-case for this information (as mentioned in the non-normative section) is troubleshooting and monitoring. For this reason, the provider name is probably a better choice.

Events added to clients themselves already provider a better means of scoping events to a client, so the `client name` wasn't helpful for this purpose.

Reply:
>  which means that an event would sometimes need to feature mutliple client names

The current implementation of JS emits the provider event through every client that it is bound to. If client 1 and client 2 both use provider 1, then client 1 and 2 will emit that event.

> Events added to clients themselves already provider a better means of scoping events to a client, so the client name wasn't helpful for this purpose.

Think so too, so to me it seems fine to include the provider name and not the client name in the event.

My only concern is, that for global events, knowing which client is affected only works by knowing which clients this provider is bound to.

Reply:
> The current implementation of JS emits the provider event through every client that it is bound to. If client 1 and client 2 both use provider 1, then client 1 and 2 will emit that event.

Yes, and that is right.

But for the corresponding global event, what happens in this case?

> My only concern is, that for global events, knowing which client is affected only works by knowing which clients this provider is bound to.

This is also true - but I think for global events, you're not so concerned with this. If you were, you'd have probably used a client event handler in the first place.

Reply:
> But for the corresponding global event, what happens in this case? Either you must fire 2 global events (that seems wrong) or the event should have both client names (which is the problem I was trying to resolve here).

Okay yes that makes sense to me.

> But for the corresponding global event, what happens in this case? Either you must fire 2 global events (that seems wrong) or the event should have both client names (which is the problem I was trying to resolve here).

I think that depends on the view. If the events would be "client events" I would call this right. If the events are "provider events" as we called them I am with you, and you should only get the event once globally.

> This is also true - but I think for global events, you're not so concerned with this. If you were, you'd have probably used a client event handler in the first place.

Makes sense to me :)

Reply:
> > My only concern is, that for global events, knowing which client is affected only works by knowing which clients this provider is bound to.
> 
> This is also true - but I think for global events, you're not so concerned with this. If you were, you'd have probably used a client event handler in the first place.

Related to [1]

To fully support named client to default provider binding as well as invoking ready handlers when registering require event handling mechanism to know about provider status 

[1] - https://github.com/open-feature/spec/pull/196/files#r1239934159

Reply:
>  require event handling mechanism to know about provider status

> While it's fun to implement Eventing spec, implementation details of this is a good example for devil is in the details idiom smile

:100: . Separation of concerns is a real challenge here. In Java, I passed around a lot of callbacks to help with much of that - so that I could maintain the exact details of some behavior to the caller, but the called object would run that delegated behavior at the right time.

The [providerRepository](https://github.com/open-feature/java-sdk/blob/cbf5c9fb581db1997450725fe6e1655a2f065dfa/src/main/java/dev/openfeature/sdk/ProviderRepository.java) for example takes callbacks that it runs on provider ready/error/shutdown etc. It has no idea what is happening within those functions (they mostly emit events, etc).